components/director/internal/destinationfetchersvc/handler_test.go

```diff
@@ -168,7 +192,31 @@ func TestHandler_FetchDestinationsSensitiveData(t *testing.T) {
 			Name:    "Missing destination query parameter.",
 			Request: reqWithUserContext,
 			DestinationFetcherSvc: func() *automock.DestinationManager {
-				return &automock.DestinationManager{}
+				svc := &automock.DestinationManager{}
+				svc.On("GetSubscribedTenantIDs", mock.Anything).Return([]string{expectedTenantID}, nil).Once()
+				return svc
+			},
+			ExpectedStatusCode: http.StatusBadRequest,
+		},
+		{
+			Name:      "Error when getting subscribed tenants fails",
+			Request:   reqWithUserContext,
+			DestQuery: namesQueryRaw,
+			DestinationFetcherSvc: func() *automock.DestinationManager {
+				svc := &automock.DestinationManager{}
+				svc.On("GetSubscribedTenantIDs", mock.Anything).Return([]string{}, testErr).Once()
+				return svc
+			},
+			ExpectedStatusCode: http.StatusBadRequest,
```

Comment:
Here 500 would be more suitable.

Reply:
This is a test and that code is expected, because it is returned at handler.go:56. Also, I'm not sure that status code 500 there is appropriate because it's a client error, not a server one.